### PR TITLE
Remove class path scanning

### PR DESCRIPTION
**Pull request by kokosing**
Remove class path scanning

### CODE REVIEW

**Comment by kokosing** [comment]
It removes:
```
2019-11-28T01:30:31.3363417Z 2019-11-28 07:15:31 INFO: Reflections took 8065 ms to scan 2 urls, producing 79 keys and 3202 values 
2019-11-28T01:31:06.3466432Z 2019-11-28 07:16:06 INFO: Reflections took 7029 ms to scan 2 urls, producing 18054 keys and 117908 values 
2019-11-28T01:31:12.6244166Z 2019-11-28 07:16:12 INFO: Reflections took 6273 ms to scan 2 urls, producing 18054 keys and 117908 values 
2019-11-28T01:31:19.0684795Z 2019-11-28 07:16:19 INFO: Reflections took 6441 ms to scan 2 urls, producing 18054 keys and 117908 values 
```

**Comment by kokosing** [comment]
So product tests should be about half minute faster per each execution.

**Review by findepi — COMMENTED** [review]
"Code cleanup" OK

File: tempto-core/src/main/java/io/prestosql/tempto/internal/fulfillment/command/SuiteCommandFulfiller.java [hunk 1]
```diff
@@ -13,8 +13,10 @@
  */
 package io.prestosql.tempto.internal.fulfillment.command;
 
+import io.prestosql.tempto.fulfillment.RequirementFulfiller;
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestosql/tempto/internal/fulfillment/command/SuiteCommandFulfiller.java:16]: > Merge bultin suite and test level fulfillers

typo

**Comment by kokosing** [comment]
Poor man benchmark.

Before:
image

After:
image

File: tempto-core/src/main/java/io/prestosql/tempto/internal/initialization/TestInitializationListener.java [hunk 2]
```diff
@@ -86,15 +88,11 @@
...
             ImmutableTablesFulfiller.class,
-            SuiteCommandFulfiller.class
-    );
-
-    private static final List<Class<? extends RequirementFulfiller>> BUILTIN_TEST_METHOD_LEVEL_FULFILLERS = ImmutableList.of(
+            SuiteCommandFulfiller.class,
             MutableTablesFulfiller.class,
-            TestCommandFulfiller.class
-    );
+            TestCommandFulfiller.class);
```
Comment (sopel39) [inline tempto-core/src/main/java/io/prestosql/tempto/internal/initialization/TestInitializationListener.java:95]: what is `TestCommandFulfiller`. Does it fit suite level fulfiller?

File: tempto-core/src/main/java/io/prestosql/tempto/fulfillment/RequirementFulfiller.java [hunk 3]
```diff
@@ -25,41 +25,21 @@
 
 public interface RequirementFulfiller
 {
-    int DEFAULT_PRIORITY = 0;
```
Comment (sopel39) [inline tempto-core/src/main/java/io/prestosql/tempto/fulfillment/RequirementFulfiller.java:28]: what is was used for? Was it used at all?

File: tempto-core/src/main/java/io/prestosql/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionsProvider.java [hunk 4]
```diff
@@ -1,122 +0,0 @@
...
-import static io.prestosql.tempto.fulfillment.table.hive.HiveTableDefinition.hiveTableDefinition;
-import static io.prestosql.tempto.fulfillment.table.jdbc.RelationalTableDefinition.relationalTableDefinition;
-import static io.prestosql.tempto.internal.convention.ConventionTestsUtils.getConventionsTestsPath;
-import static java.nio.file.Files.exists;
-import static java.nio.file.Files.newDirectoryStream;
-import static java.util.Collections.emptyList;
-import static java.util.stream.Collectors.toList;
-import static org.slf4j.LoggerFactory.getLogger;
-
-public class ConventionTableDefinitionsProvider
```
Comment (sopel39) [inline tempto-core/src/main/java/io/prestosql/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionsProvider.java:42]: undo? Why to inline it?

File: tempto-core/src/main/java/io/prestosql/tempto/context/TestContext.java [hunk 5]
```diff
@@ -75,6 +75,8 @@ default TestContext createChildContext(State... states)
      */
```
Comment (sopel39) [inline tempto-core/src/main/java/io/prestosql/tempto/context/TestContext.java:75]: wow. This is quite sophisticated commit for what it does. I would expect simple `if` somewhere to NOOP on empty requirements.

Why do ne need additional method like ` void injectMembers(Object instance);`

File: README.md [hunk 6]
```diff
@@ -55,7 +55,6 @@ each test and suite fulfillment process to separate files. To use that configure
 log4j.rootLogger=INFO, TEST_FRAMEWORK_LOGGING_APPENDER
```
Comment (sopel39) [inline README.md:55]: is it breaking change in relation to Presto?

File: tempto-core/src/main/java/io/prestosql/tempto/internal/initialization/TestInitializationListener.java (hunk 2 above)
Comment (kokosing) [reply tempto-core/src/main/java/io/prestosql/tempto/internal/initialization/TestInitializationListener.java:95]: It runs a command before or after each test. No it is test level fullfiler.

File: README.md (hunk 6 above)
Comment (kokosing) [reply README.md:55]: I didn't notice any regression. I tested it locally.

File: tempto-core/src/main/java/io/prestosql/tempto/context/TestContext.java (hunk 5 above)
Comment (kokosing) [reply tempto-core/src/main/java/io/prestosql/tempto/context/TestContext.java:75]: > Why do ne need additional method like void injectMembers(Object instance);

To not to rely on GuiceTestContext (TestContext implementation). So I could use `ForwardingTestContext`

File: tempto-core/src/main/java/io/prestosql/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionsProvider.java (hunk 4 above)
Comment (kokosing) [reply tempto-core/src/main/java/io/prestosql/tempto/internal/convention/tabledefinitions/ConventionTableDefinitionsProvider.java:42]: I wanted to have a better (and clear) control how `TABLE_DEFINITIONS_REPOSITORY` is used and where so I could control how we inject additional tables.

**Comment by kokosing** [comment]
@sopel39 I responded to comments.

File: tempto-runner/src/main/java/io/prestosql/tempto/runner/TemptoRunner.java [hunk 7]
```diff
@@ -62,8 +85,20 @@ public static void runTempto(TemptoRunnerCommandLineParser parser, String[] args
...
+    public static void runTempto(
+            TemptoRunnerCommandLineParser parser,
+            TemptoRunnerOptions options,
+            Supplier<List<Class<? extends TestMethodModuleProvider>>> additionalTestModules,
+            Supplier<List<Class<? extends SuiteModuleProvider>>> additionalSuiteModules,
+            Supplier<List<Class<? extends RequirementFulfiller>>> additionalFulfillersSupplier,
+            Supplier<List<Class<? extends TableManager>>> additionalTableManagers,
+            Supplier<List<TableDefinition>> additionalTableDefinitions)
     {
+        TestInitializationListener.ADDITIONAL_TEST_MODULES = additionalTestModules;
```
Comment (sopel39) [inline tempto-runner/src/main/java/io/prestosql/tempto/runner/TemptoRunner.java:97]: static import?

**Review by sopel39 — APPROVED** [review]
make sure it works with Presto

**Comment by sopel39** [comment]
nit: add super easy benchmark maybe.

File: tempto-runner/src/main/java/io/prestosql/tempto/runner/TemptoRunner.java (hunk 7 above)
Comment (kokosing) [reply tempto-runner/src/main/java/io/prestosql/tempto/runner/TemptoRunner.java:97]: I think this way is more readable.

**Comment by kokosing** [comment]
> nit: add super easy benchmark maybe.

Presto is the best benchmark ;)

**Pull request merged**